Pass python_requires argument to setuptools

jdufresne opened the pull request:
Helps pip decide what version of the library to install.

https://packaging.python.org/tutorials/distributing-packages/#python-requires

> If your project only runs on certain Python versions, setting the python_requires argument to the appropriate PEP 440 version specifier string will prevent pip from installing the project on other Python versions.

https://setuptools.readthedocs.io/en/latest/setuptools.html#new-and-changed-setup-keywords

> python_requires
>
> A string corresponding to a version specifier (as defined in PEP 440) for the Python version, used to specify the Requires-Python defined in PEP 345.

jdufresne committed:
Pass python_requires argument to setuptools

Helps pip decide what version of the library to install.

https://packaging.python.org/tutorials/distributing-packages/#python-requires

> If your project only runs on certain Python versions, setting the
> python_requires argument to the appropriate PEP 440 version specifier
> string will prevent pip from installing the project on other Python
> versions.

https://setuptools.readthedocs.io/en/latest/setuptools.html#new-and-changed-setup-keywords

> python_requires
>
> A string corresponding to a version specifier (as defined in PEP 440)
> for the Python version, used to specify the Requires-Python defined in
> PEP 345.

tk0miya commented:
>Do not allow installing on Python 3.5.1 as typing.TYPE_CHECKING is not available.

That is simply bug. Not new dependencies.

jdufresne commented:
> That is simply bug. Not new dependencies.

I see, thanks. I have removed the 3.5.1 check as suggested and updated the PR.

I'll create a compat shim for `TYPE_CHECKING` as a separate PR.

tk0miya reviewed (approved):
LGTM!

To be available on next release, please rebase this onto 1.7 branch.
Thanks,

jdufresne commented:
Done! Rebased on 1.7.

tk0miya commented:
Merged. Thank you for quick work!

tk0miya commented:
About `TYPE_CHECKING`, I'm working on #4735. So please wait a moment.
Thanks,
